surfacers/datadog/datadog.go

```diff
@@ -0,0 +1,196 @@
+// Copyright 2017 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package datadog
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"regexp"
+	"time"
+
+	datadog "github.com/DataDog/datadog-api-client-go/api/v1/datadog"
+	"github.com/google/cloudprober/logger"
+	"github.com/google/cloudprober/metrics"
+	"github.com/google/cloudprober/surfacers/common/options"
+	configpb "github.com/google/cloudprober/surfacers/datadog/proto"
+	"google.golang.org/protobuf/proto"
+)
+
+/*
+	The datadog surfacer presents EventMetrics to the datadog SubmitMetrics APIs,
+	using the config passed in.
+
+	Some EventMetrics are not supported here, as the datadog SubmitMetrics API only
+	supports float64 type values as the metric value.
+*/
+
+// Datadog API limit for metrics included in a SubmitMetrics call
+const datadogMaxSeries int = 20
+
+// All metrics are of type gauge
+const gauge string = "gauge"
```

Comment:
I am still reviewing the recent changes, but I was wondering why all metrics have to be of "gauge" type. Datadog API seems to allow both.

Also, even though you're exporting them with type set to gauge, these metrics are all cumulative (counters) by default. Ideally you'd choose the type based on EventMetrics.Kind.